units/provisioning_agent_deploy.go

```diff
@@ -251,7 +251,14 @@ func (j *agentDeployJob) prepRemoteHost(ctx context.Context, hostObj host.Host,
 	if hostObj.Distro.Setup == "" {
 		return nil
 	}
+
 	if logs, err := hostObj.RunSSHCommand(ctx, hostObj.SetupCommand(), sshOptions); err != nil {
+		if strings.Contains(logs, fmt.Sprintf("mv: cannot stat '%s': No such file or directory", evergreen.TempSetupScriptName)) ||
```

Comment:
The only way I could check if mv failed due to a different SSH command running the setup script was by checking for specific strings. I don't know if there's a cleaner way to do this.